src/TimeManager.vala

```diff
@@ -0,0 +1,41 @@
+public class Maya.TimeManager : Object {
+    private static TimeManager instance = null;
+    private uint timeout_id = 0;
+
+    public signal void on_update_today ();
+
+    public TimeManager () {
+        setup_today_callback ();
+    }
+
+    private bool callback_update_today () {
+        debug ("Updating today");
+        on_update_today ();
+        setup_today_callback ();
+        return false;
+    }
+
+    private void setup_today_callback () {
+        debug ("Setting new callback to update today");
+        if (timeout_id > 0) {
+            GLib.Source.remove (timeout_id);
+            timeout_id = 0;
+        }
+
+        // Sets a timeout that will call at the end of the day.
+        // Creates a timer for the time until end of day, and adds an extra
+        // second to be sure it's called after midnight.
+        var now = new DateTime.now_local ();
+        var tomorrow = Util.strip_time (now.add_days (1));
+        var interval = (tomorrow.difference (now) + 1) / 1000000;
+        timeout_id = GLib.Timeout.add_seconds ((uint) interval, callback_update_today);
```

Comment:
Does this work if the computer is suspended/hibernated?

Reply:
Nope, thanks for thinking of that. Working on it, and thanks for the suggestion.

Reply:
Ok, I added functionality to deal with this. It updates the date and sets up a new callback when the computer resumes from sleep. To the best of my knowledge, it didn't change the normal operation and sets up a proper (regular) callback when it's done.

I also have some work to update faster when settings are being changed ([like in Wingpanel too](https://github.com/elementary/wingpanel-indicator-datetime/blob/fb6281eb0302121f503e2f2fcba268e5e6294b92/src/Services/TimeManager.vala#L63)) but I think that may be another PR since it may involve some design work.